extras/rya.pcj.fluo/pcj.fluo.app/src/main/java/org/apache/rya/indexing/pcj/fluo/app/JoinResultUpdater.java

```diff
@@ -408,38 +408,28 @@ public boolean hasNext() {
         }
 
         @Override
-        public BindingSet next() {
+        public VisibilityBindingSet next() {
             final MapBindingSet bs = new MapBindingSet();
 
             for(final Binding binding : newResult) {
                 bs.addBinding(binding);
             }
 
-            final BindingSet joinResult = joinedResults.next();
+            final VisibilityBindingSet joinResult = joinedResults.next();
             for(final Binding binding : joinResult) {
                 bs.addBinding(binding);
             }
 
             String visibility = "";
-            if(newResult instanceof VisibilityBindingSet && joinResult instanceof VisibilityBindingSet) {
-                final Joiner join = Joiner.on(")&(");
-                final String leftVisi = ((VisibilityBindingSet)newResult).getVisibility();
-                final String rightVisi = ((VisibilityBindingSet)joinResult).getVisibility();
-                if(leftVisi.isEmpty() || rightVisi.isEmpty()) {
-                    visibility = (leftVisi + rightVisi).trim();
-                } else {
-                    visibility = "(" + join.join(leftVisi, rightVisi) + ")";
-                }
-                return new VisibilityBindingSet(bs, visibility);
-            } else if(newResult instanceof VisibilityBindingSet) {
-                visibility = ((VisibilityBindingSet) newResult).getVisibility();
-                return new VisibilityBindingSet(bs, visibility);
-            } else if(joinResult instanceof VisibilityBindingSet) {
-                visibility = ((VisibilityBindingSet) joinResult).getVisibility();
-                return new VisibilityBindingSet(bs, visibility);
+            final Joiner join = Joiner.on(")&(");
+            final String leftVisi = newResult.getVisibility();
+            final String rightVisi = joinResult.getVisibility();
+            if(leftVisi.isEmpty() || rightVisi.isEmpty()) {
+                visibility = (leftVisi + rightVisi).trim();
             } else {
-                return new VisibilityBindingSet(bs);
+                visibility = "(" + join.join(leftVisi, rightVisi) + ")";
```

Comment:
Using a joiner here is kind of overkill since you're only ever joining two items.